compiler/stable_mir/src/lib.rs

```diff
@@ -52,10 +54,27 @@ impl Debug for DefId {
     }
 }
 
+impl IndexToVal for DefId {
+    fn to_val(index: usize) -> Self
+    where
+        Self: std::marker::Sized,
+    {
+        DefId(index)
+    }
+}
```

Comment:
If you also add the opposite direction, you can make all the `pub usize` fields private again

Reply:
And in that case you can rename the trait just to `IndexedVal` or sth that is direction agnostic